appinventor/components/src/com/google/appinventor/components/runtime/Spreadsheet.java

```diff
@@ -661,6 +653,54 @@ public void FinishedWriteRow () {
     EventDispatcher.dispatchEvent(this, "FinishedWriteRow");
   }
 
+  @SimpleFunction(description = "Adds a new sheet inside the Spreadsheet")
+  public void AddSheet(final String sheetName){
+    if (spreadsheetID == null || spreadsheetID.isEmpty()) {
+      ErrorOccurred("AddSheet: " + "SpreadsheetID is empty.");
+      return;
+    } else if (credentialsPath == null || credentialsPath.isEmpty()) {
+      ErrorOccurred("AddSheet: " + "Credentials JSON is required.");
+      return;
+    }
+    // Run the API call asynchronously
+    AsynchUtil.runAsynchronously(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          Sheets sheetsService = getSheetsService();
+          AddSheetRequest addSheetRequest = new AddSheetRequest()
+                  .setProperties(new SheetProperties()
+                          //set title of new sheet as user parameter
+                          .setTitle(sheetName)
+                  );
+          List<Request> requests = new ArrayList<>();
+          requests.add(new Request().setAddSheet(addSheetRequest));
+          BatchUpdateSpreadsheetRequest body = new BatchUpdateSpreadsheetRequest()
+                  .setRequests(requests);
+          sheetsService.spreadsheets().batchUpdate(spreadsheetID, body).execute();
+
+          // Run the callback event block
+          activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+              FinishedAddSheet(sheetName);
+            }
+          });
+
+        }
+        catch (Exception e) {
+          e.printStackTrace();
+          ErrorOccurred("AddSheet: " + e.getMessage());
+        }
+      }
+    });
+  }
+  @SimpleEvent(
+          description="The callback event for the addSheet block, called once the " +
+                  "values on the table have been updated.")
+  public void FinishedAddSheet(final String sheetName) {
```

Comment:
I think SheetAdded is slightly less wordy but I can see the pattern you are following. @SusanRatiLane may want to comment.